src/h5web/visualizations/shared/utils.ts

```diff
@@ -1,36 +1,32 @@
 import { scaleLinear } from 'd3-scale';
 import { extent } from 'd3-array';
-import { AxisOffsets, Size, Domain } from './models';
+import { Size, Domain } from './models';
 
 export const adaptedNumTicks = scaleLinear()
   .domain([300, 900])
   .rangeRound([3, 10])
   .clamp(true);
 
 export function computeVisSize(
-  visAreaSize: Size,
-  axisOffsets: AxisOffsets,
+  availableSize: Size,
   aspectRatio?: number
 ): Size | undefined {
-  const { width, height } = visAreaSize;
+  const { width, height } = availableSize;
 
-  if (width === 0 && height === 0) {
+  if (width <= 0 && height <= 0) {
```

Comment:
🤔 don't think these can be negative ^^ - `0` is the special case for `useMeasure`'s return value.

Reply:
Yes but the dimensions here are the ones with the padding subtracted. Is `useThree` always taking the padding into account ?

Reply:
Ahh well, then that's why computing `availableWidth` inside the function, after this first return, made sense.

Reply:
Otherwise, you risk not rendering anything if the rendering area is less than or equal to the padding.

Reply:
It is fine for me :upside_down_face:

Reply:
You're right, I got confused somewhere -- if there's no space available for the canvas, no point in rendering it... 🤦